### ISSUE TITLE
[Java] Deserialization fails when serializer is package protected

### ISSUE THREAD

**Issue by knutwannheden** [alipay/fury#998]
Complete test case:

```java
package test;

import io.fury.Fury;
import io.fury.memory.MemoryBuffer;
import io.fury.serializer.Serializer;

import java.util.UUID;

public class FuryTest {

    public static void main(String[] args) {
        Fury fury = Fury.builder()
                .withRefTracking(true)
                .requireClassRegistration(false)
                .build();
        fury.registerSerializer(UUID.class, new UUIDSerializer(fury));

        DomainObject obj = new DomainObject();
        obj.id = UUID.randomUUID();
        byte[] serialized = fury.serialize(obj);

        DomainObject obj2 = (DomainObject) fury.deserialize(serialized);
    }
}

class DomainObject {
    UUID id;
}

class UUIDSerializer extends Serializer<UUID> {
    public UUIDSerializer(Fury fury) {
        super(fury, UUID.class);
    }

    @Override
    public UUID read(MemoryBuffer buffer) {
        return new UUID(buffer.readLong(), buffer.readLong());
    }

    @Override
    public void write(MemoryBuffer buffer, UUID value) {
        buffer.writeLong(value.getMostSignificantBits());
        buffer.writeLong(value.getLeastSignificantBits());
    }
}
```

Declaring the `UUIDSerializer` as public (nested or top-level) fixes the issue.

**Comment by chaokunyang** [comment]
I can reproduce it:
```java
Exception in thread "main" java.lang.IllegalAccessError: failed to access class io.fury.UUIDSerializer from class io.fury.DomainObjectFuryRefCodec_31953163 (io.fury.UUIDSerializer is in unnamed module of loader 'app'; io.fury.DomainObjectFuryRefCodec_31953163 is in unnamed module of loader io.fury.util.ClassLoaderUtils$ByteArrayClassLoader @2974f221)
	at io.fury.DomainObjectFuryRefCodec_31953163.<init>(DomainObjectFuryRefCodec_31953163.java:40)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at io.fury.serializer.Serializers.newSerializer(Serializers.java:74)
	at io.fury.resolver.ClassResolver.createSerializer(ClassResolver.java:1154)
	at io.fury.resolver.ClassResolver.getOrUpdateClassInfo(ClassResolver.java:1095)
	at io.fury.Fury.writeRef(Fury.java:283)
	at io.fury.Fury.serialize(Fury.java:235)
	at io.fury.Fury.serialize(Fury.java:183)
```

Seems it's related to java module?

**Comment by chaokunyang** [comment]
It's a different mistake on JDK17+, after fix this, I can get your exception:
```java
Exception in thread "main" java.lang.IllegalAccessError: failed to access class io.fury.UUIDSerializer from class io.fury.DomainObjectFuryRefCodec_31953163 (io.fury.UUIDSerializer is in unnamed module of loader 'app'; io.fury.DomainObjectFuryRefCodec_31953163 is in unnamed module of loader io.fury.util.ClassLoaderUtils$ByteArrayClassLoader @2974f221)
	at io.fury.DomainObjectFuryRefCodec_31953163.<init>(DomainObjectFuryRefCodec_31953163.java:40)
	at io.fury.serializer.Serializers.newSerializer(Serializers.java:79)
	at io.fury.resolver.ClassResolver.createSerializer(ClassResolver.java:1154)
	at io.fury.resolver.ClassResolver.getOrUpdateClassInfo(ClassResolver.java:1095)
	at io.fury.Fury.writeRef(Fury.java:283)
	at io.fury.Fury.serialize(Fury.java:235)
	at io.fury.Fury.serialize(Fury.java:183)
```

**Comment by chaokunyang** [comment]
Hi, we've just released fury 0.2.1: https://github.com/alipay/fury/releases/tag/v0.2.1, it should fixed this issue. Could you try it out?
